src/download.rs

```diff
@@ -81,13 +85,13 @@ pub fn download_file(url: &str, file_path: &Path, redirect_path: &Path) -> Resul
           measurements.pop_front();
         }
         let avg_speed = measurements.iter().sum::<f64>() / measurements.len() as f64;
-        let eta = if avg_speed > 0.0 {
+        let eta = if avg_speed > 1.0 {
```

Comment:
Is ETA zero if the speed is slow (less than 1.0)? Perhaps `eta` should be an optional with None meaning "can't tell"?

Or something like this for clarity:

```rust 
enum Eta {
    Unknown,
    Seconds(usize),
}

impl std::fmt::Display for Eta {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            Eta::Unknown => write!(f, "unknown"),
            Eta::Seconds(s) => write!(f, "{s}s"),
        }
    }
}
```